src/main/java/nl/beehivejava/scriptj/io/ResourceLoader.java

```diff
@@ -0,0 +1,24 @@
+package nl.beehivejava.scriptj.io;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * A {@code ResourceLoader} can convert a location (as a {@link String}) to an {@link InputStream}. For example: a
+ * {@code UrlResourceLoader} could be implemented, loading an URL (as a {@code String}) to an {@code InputStream}.
+ *
+ * @author Lesley
+ */
+public interface ResourceLoader {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Annotate the "ResourceLoader" interface with the @FunctionalInterface annotation (sonar.java.source not set. Assuming 8 or greater.) [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.tehmessiah.nl/coding_rules#rule_key=squid%3AS1609)